scoper.php

```diff
@@ -71,6 +71,17 @@
             // comment out
             str_replace('\\' . $prefix . '\trigger_deprecation(', '// \trigger_deprecation(', $content),
 
+        // make external rules easier to write without enforing getRuleDefinition()
+        // as they are not designed for open-sourcing
+        static function (string $filePath, string $prefix, string $content): string {
+            if (!\str_ends_with($filePath, 'vendor/symplify/rule-doc-generator-contracts/src/Contract/DocumentedRuleInterface.php')) {
+                return $content;
+            }
+
+            // comment out
+            return str_replace('public function getRuleDefinition', '// public function getRuleDefinition', $content);
```

Comment:
@TomasVotruba instead of commenting out, I think it is better to have default empty getRuleDefinition in AbstractRector, eg:

```php
abstract class AbstractRector extends NodeVisitorAbstract implements RectorInterface
{
    public function getRuleDefinition(): RuleDefinition
    {
        return new RuleDefinition('', []);
    }
}
```

wdyt? It doesn't enforce user to create it, but can create if they want to :)